k8s/supportability/src/collect/logs/mod.rs

```diff
@@ -137,6 +137,28 @@ impl LogCollection {
 
             let mut hostname = None;
             if is_host_name_required(service_name.clone()) {
+                let status = pod.status.clone().ok_or_else(|| {
+                    K8sResourceError::invalid_k8s_resource_value(format!(
```

Comment:
This could still potentially throw an error if we the pod is being recreated, and we catch it without a state.
I suggest you do this bit where the second function is per-pod on a separate function, and then we can catch any error, example:

```
for pod in pods {
    match self.pod_logging_resources() {
        Ok(resources) => logging_resources.extend(resources),
        Err(error) => log(format!(
                        "Skipping the pod in pending state: {:?}",
                        pod.metadata.name
                    )),
    }
}
```

I suspect we may also have this type of issue elsewhere in the support bundle code, probably we need to analyse the rest of the code at some point.

Reply:
done